pypeit/find_objects.py

```diff
@@ -0,0 +1,1330 @@
+        skymask_now = skymask if (skymask is not None) else np.ones_like(self.sciImg.image, dtype=bool)
+
+        # Allow for previous sky to better estimate ivar
+        #  Unless we used a background image (i.e. bkg_redux=True)
+        if (previous_sky is not None) and (not self.bkg_redux):
+            # Estimate the variance using the input sky model
+            var = procimg.variance_model(self.sciImg.base_var,
+                                          counts=previous_sky,
+                                          count_scale=self.sciImg.img_scale,
+                                          noise_floor=self.sciImg.noise_floor)
+            self.sciImg.ivar = utils.inverse(var)
+
+        # Loop on slits
+        for slit_idx in gdslits:
+            slit_spat = self.slits.spat_id[slit_idx]
+            msgs.info("Global sky subtraction for slit: {:d}".format(slit_spat))
+            thismask = self.slitmask == slit_spat
+            inmask = self.sciImg.select_flag(invert=True) & thismask & skymask_now
+            # All masked?
+            if not np.any(inmask):
+                msgs.warn("No pixels for fitting sky.  If you are using mask_by_boxcar=True, your radius may be too large.")
+                self.reduce_bpm[slit_idx] = True
+                continue
+
+            # Find sky
+            global_sky[thismask] = skysub.global_skysub(
+                self.sciImg.image, self.sciImg.ivar, self.tilts, thismask,
+                self.slits_left[:,slit_idx], self.slits_right[:,slit_idx],
+                inmask=inmask, sigrej=sigrej,
+                bsp=self.par['reduce']['skysub']['bspline_spacing'],
+                no_poly=self.par['reduce']['skysub']['no_poly'],
+                pos_mask=(not self.bkg_redux), show_fit=show_fit)
+
+            # Mask if something went wrong
+            if np.sum(global_sky[thismask]) == 0.:
+                msgs.warn("Bad fit to sky.  Rejecting slit: {:d}".format(slit_idx))
+                self.reduce_bpm[slit_idx] = True
+
+        if update_crmask and self.par['scienceframe']['process']['mask_cr']:
+            # Find CRs with sky subtraction
+            # TODO: Shouldn't the saturation flagging account for the
+            # subtraction of the sky?
+            self.sciImg.build_crmask(self.par['scienceframe']['process'],
+                                     subtract_img=global_sky)
+            # Update the fullmask
+            self.sciImg.update_mask_cr(self.sciImg.crmask)
+
+        # Step
+        self.steps.append(inspect.stack()[0][3])
+
+        if show:
+            sobjs_show = None if show_objs else sobjs_obj
+            # Global skysub is the first step in a new extraction so clear the channels here
+            self.show('global', slits=True, sobjs=sobjs_obj, clear=False)
+
+        # Return
+        return global_sky
+
+
+    # TODO This method only used for IFUs, so it should be present in the IFU subclass not here.
```

Comment:
Actually, it's not specific to IFU. It allows a user to define the sky regions manually, even if that's longslit or multislit. I'd recommend keeping it here.